packages/babel-parser/src/tokenizer/index.js

```diff
@@ -439,12 +452,12 @@ export default class Tokenizer extends LocationParser {
   readToken_interpreter(): boolean {
     if (this.state.pos !== 0 || this.length < 2) return false;
 
+    let ch = this.input.charCodeAt(this.state.pos + 1);
+    if (ch !== charCodes.exclamationMark) return false;
+
     const start = this.state.pos;
     this.state.pos += 1;
 
-    let ch = this.input.charCodeAt(this.state.pos);
-    if (ch !== charCodes.exclamationMark) return false;
```

Comment:
What is this change for?

Reply:
I moved this because it will incorrectly eat a `#` character if the next character isn't a `!`, I think that the `this.state.pos` should only be advanced if it is actually going to be an interpreter directive. (This happens if the first line of the file is `#{}`